include/drivers/rtc.h

```diff
@@ -0,0 +1,262 @@
+/**
+ * @file
+ * @brief RTC public API header file.
+ */
+/*
+ * Copyright (c) 2021 LACROIX Group
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_RTC_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <time.h>
+#include <device.h>
+// #include <dt-bindings/rtc/rtc.h>
+
+/**
+ * @brief RTC driver APIs
+ * @defgroup rtc_interface RTC driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/** @brief Wake up timers enum. */
+enum rtc_wakeup_id {
+	RTC_WUT0,       /**< Wake Up timer 0. */
+	/* Add new item before RTC_WUT_NUM. */
+	RTC_WUT_NUM,    /**< Number of wut for the driver. */
+};
+
+/** @brief Alarms enum. */
+enum rtc_alarm_id {
+	RTC_ALARMA,     /**< Alarm A. */
+	RTC_ALARMB,     /**< Alarm B. */
+	/* Add new item before RTC_ALARM_NUM. */
+	RTC_ALARM_NUM,  /**< Number of alarms for the driver. */
+};
+
+/** @brief Specifies if the RTC Alarm is on date or Week day. */
+enum rtc_alarm_date_weekday {
+	RTC_ALARM_DATE_SEL,     /**< Alarm Date is selected, set parameter to a value in the 1-31 range. */
+	RTC_ALARM_WEEKDAY_SEL,  /**< Alarm Weekday is selected, set parameter to a value in the 0-6 range (days since Sunday). */
+};
+
+/** @brief Specifies the RTC Alarm Masks. */
```

Comment:
A bit more specificity about what this mask means exactly would be useful.
I'm looking at RTC functionality for the [MAX31343](https://datasheets.maximintegrated.com/en/ds/MAX31343.pdf) as a possible use case for this API.

That RTC has the option to mask a variety of time fields (from seconds up to years), but only some combinations are valid. For example, you cannot mask to match on minutes but not on seconds. I _think_ that's implied with this enum but I am not sure without more context.

Reply:
I test this driver with an internal rtc of a STM32 µC (an example was committed).
For this rtc you can add a mask on
 - secondes => RTC_ALARM_MASK_SEC
 - minutes => RTC_ALARM_MASK_MIN
 - hours => RTC_ALARM_MASK_HOURS
 - date or weekday => RTC_ALARM_MASK_DATE_WEEKDAY
 
 If you select RTC_ALARM_MASK_MIN the alarm will occur every minutes
 if you select RTC_ALARM_MASK_DATE_WEEKDAY the alarm will occur
  - every sunday with alarm_time.tm_wday = 0 and use  alarm_date_weekday_sel=RTC_ALARM_WEEKDAY_SEL
  - every monday with alarm_time.tm_wday = 1 and use  alarm_date_weekday_sel=RTC_ALARM_WEEKDAY_SEL
  ...
  - every 2nd of month with alarm_time.tm_mday = 2 and use  alarm_date_weekday_sel=RTC_ALARM_DATE_SEL